learn-arai/app/hooks/useAuth.ts

```diff
@@ -0,0 +1,79 @@
+'use client';
+
+// this react hook responsible for checking if user is logged in.
+import { redirect } from 'next/navigation';
+
+import { useEffect, useState } from 'react';
+
+import { User } from './useUser';
+import { useUser } from './useUser';
+
+export const useAuth = () => {
+    const { addUser, removeUser, user, fetchedUser, getUserFromLocalStorage } =
+        useUser();
+    const [isAuthenticated, setIsAuthenticated] = useState<boolean>();
+
+    useEffect(() => {
+        checkSession();
+    });
+
+    const signIn = async (credentials: FormData) => {
+        const response = await sendCredentialToServer(credentials);
+        const email = await fetchedUser();
```

Comment:
Shouldn't signIn return needed data already, why bother call API twice?

Reply:
As I said above make `/sign-in` route return required fields, so we don't have to fetch twice